spec/controllers/admin/books_controller_spec.rb

```diff
@@ -0,0 +1,110 @@
+require "rails_helper"
+include SessionsHelper
+RSpec.describe Admin::BooksController, type: :controller do
+  let(:admin1){FactoryBot.create :user, role: "admin"}
+  let!(:book_3){FactoryBot.create :book}
+  let!(:book_4){FactoryBot.create :book}
```

Comment:
Sao lại cần phải dùng let! ở đây em

Reply:
vì dùng let! để lưu lại 1 đối tượng vào database, như thế các khối lệnh it đằng sau đều có thể sử dụng đc, không dùng let ở đây được vì nó chỉ được khởi chạy sau 1 khối lệch it và các khối lệnh it bên trong thôi ạ